### PR TITLE
docs(splunk_hec source): Use generated docs for configuration 

### LINKED ISSUE #16078: Convert the `splunk_hec` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #16078 
Epic #14998

### CODE REVIEW

File: website/cue/reference/components/sources/splunk_hec.cue [hunk 1]
```diff
@@ -51,93 +52,8 @@ components: sources: splunk_hec: {
...
-			}
-		}
-		store_hec_token: {
-			common:      false
-			description: "When incoming requests contain a Splunk HEC token, if this setting is set to `true`, the token will kept in the event metadata and will be used if the event is sent to a Splunk HEC sink."
-			required:    false
-			type: bool: default: false
-		}
+	configuration: base.components.sources.splunk_hec.configuration & {
+		token: warnings: ["This option has been deprecated, the `valid_tokens` option should be used."]
```
Comment (spencergilbert) [inline website/cue/reference/components/sources/splunk_hec.cue:56]: I don't think we've consistently been using a `warning` for deprecated options - is that something we should be doing?

**Review by spencergilbert — APPROVED** [review]

File: website/cue/reference/components/sources/splunk_hec.cue (hunk 1 above)
Comment (StephenWakely) [reply website/cue/reference/components/sources/splunk_hec.cue:56]: It wasn't consistent in the old docs either. This is the first time I've come across one that included it as a warning. I quite like it though, and vote we do it throughout. Maybe chat in the stand up about it?

Comment (neuronull) [reply website/cue/reference/components/sources/splunk_hec.cue:56]: 👍 to the idea.

File: website/cue/reference/components/sources/splunk_hec.cue [hunk 2]
```diff
@@ -51,93 +52,8 @@ components: sources: splunk_hec: {
...
-			}
-		}
-		token: {
-			common:      true
-			description: "If supplied, incoming requests must supply this token in the `Authorization` header, just as a client would if it was communicating with the Splunk HEC endpoint directly. If _not_ supplied, the `Authorization` header will be ignored and requests will not be authenticated."
-			required:    false
-			warnings: ["This option has been deprecated, the `valid_tokens` option should be used."]
-			type: string: {
-				default: null
-				examples: ["A94A8FE5CCB19BA61C4C08"]
```
Comment (neuronull) [inline website/cue/reference/components/sources/splunk_hec.cue:120]: Looks like this example wasn't carried over.

**Review by neuronull — CHANGES_REQUESTED** [review]

File: website/cue/reference/components/sources/splunk_hec.cue (hunk 2 above)
Comment (StephenWakely) [reply website/cue/reference/components/sources/splunk_hec.cue:120]: > Looks like this example wasn't carried over.

Do we need examples for deprecated options?

Comment (neuronull) [reply website/cue/reference/components/sources/splunk_hec.cue:120]: oh good point. I missed that. I don't think we do

**Review by neuronull — APPROVED** [review]

**Pull request merged**